include/shuffle/ScatterShuffle.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <thrust/device_vector.h>
+#include <thrust/scatter.h>
+#include <thrust/random/linear_congruential_engine.h>
+#include <thrust/random/uniform_int_distribution.h>
+#include "DefaultRandomGenerator.h"
+#include "shuffle/Shuffle.h"
+
+//Writes into random addresses
+// Doesn't actually shuffle, only used for measurement purposes
+template <class BijectiveFunction, class ContainerType = thrust::device_vector<uint64_t>, class RandomGenerator = DefaultRandomGenerator>
+class ScatterShuffle : public Shuffle<ContainerType, RandomGenerator>
+{
+public:
+    void shuffle( const ContainerType& in_container, ContainerType& out_container, uint64_t seed, uint64_t num ) override
+    {
+        thrust::counting_iterator<uint64_t>   counting( 0 );
+auto scatter_key= thrust::make_transform_iterator(counting, [=]__device__ (uint64_t idx){
+thrust::minstd_rand rng(seed);
+rng.discard(idx);
+
+  thrust::uniform_int_distribution<uint64_t> dist(0,num-1);
+  return dist(rng);
+});
+thrust::scatter(in_container.begin(),in_container.end(),scatter_key,out_container.begin());

```

Comment:
I have clang format set up to ensure consistent formatting if you want. Not sure if you have it set up on w/e IDE you are using though

Reply:
I've been running clans format command line on these files, looks like I missed some though.

Reply:
@RAMitchell `clang-format -i -style=file $(find -path ./vendor -prune -o -regex ".*\.\(cu\|h\)")` command will format everything when run from the base directory